packages/template-block/src/TemplateBlock.tsx

```diff
@@ -0,0 +1,233 @@
+        templateDescription: description ?? '',
+    });
+
+    const hasPreview = preview !== PreviewType.None;
+    const flexDirection = hasPreview ? textPositioningToFlexDirection[textPositioning] : 'row';
+    const isRows = hasPreview && (flexDirection === 'row' || flexDirection === 'row-reverse');
+    const borderRadius = hasRadius_blockCard ? radiusValue_blockCard : radiusStyleMap[radiusChoice_blockCard];
+    const border = hasBorder_blockCard
+        ? `${borderWidth_blockCard} ${borderStyle_blockCard} ${toRgbaString(borderColor_blockCard)}`
+        : 'none';
+
+    useEffect(() => {
+        updateBlockSettings({ title: templateTitle });
+    }, [templateTitle]);
+
+    useEffect(() => {
+        updateBlockSettings({ description: templateDescription });
+    }, [templateDescription]);
+
+    useEffect(() => {
+        if (error !== null) {
+            setLastErrorMessage(error);
+            dispatch({ type: TemplateDataActionType.UPDATE_TITLE, payload: { newValue: '' } });
+            dispatch({ type: TemplateDataActionType.UPDATE_DESCRIPTION, payload: { newValue: '' } });
+        }
+    }, [error]);
+
+    useEffect(() => {
+        if (blockTemplates[TEMPLATE_BLOCK_SETTING_ID] && blockTemplates[TEMPLATE_BLOCK_SETTING_ID].length > 0) {
+            const lastTemplate = blockTemplates[TEMPLATE_BLOCK_SETTING_ID].pop();
+
+            if (lastTemplate) {
+                setSelectedTemplate(lastTemplate);
+            }
+        }
+    }, [blockTemplates]);
+
+    const handleNewPublication = () => {
+        if (selectedTemplate !== null) {
+            if (Array.isArray(previewCustom) && previewCustom.length > 0) {
+                selectedTemplate.previewUrl = previewCustom[0].previewUrl;
+            }
+
+            const options: OpenNewPublicationPayload = {
+                template: selectedTemplate,
+            };
+
+            appBridge.dispatch(openNewPublication(options));
+        }
+    };
+
+    const onTemplateSelected = async (result: TemplateLegacy) => {
+        const { title, description } = blockSettings;
+
+        try {
+            await updateTemplateIdsFromKey(TEMPLATE_BLOCK_SETTING_ID, [result.id]);
+            updateBlockSettings({
+                template: result,
+                templateId: result.id,
+            });
+            dispatch({
+                type: TemplateDataActionType.UPDATE_TITLE,
+                payload: { newValue: result.title, prevValue: title },
+            });
+            dispatch({
+                type: TemplateDataActionType.UPDATE_DESCRIPTION,
+                payload: { newValue: result.description, prevValue: description },
+            });
+        } catch (error) {
+            setLastErrorMessage(error as string);
+        }
+
+        appBridge.closeTemplateChooser();
+    };
+
+    const handleOpenTemplateChooser = () => appBridge.openTemplateChooser(onTemplateSelected);
+
+    return (
+        <div data-test-id="template-block-container">
```

Comment:
```suggestion
        <div data-test-id="template-block-container" classname="template-block">
```

You need to add this one for the scoping ;)

Reply:
True. Sorry, I've looked up for a `template-block` value in a code and found a `data-test-id` with the same value. 
Anyway, fixed. Thanks for pointing out!